kedro-datasets/kedro_datasets/xarray/geotiff_dataset.py

```diff
@@ -0,0 +1,111 @@
+"""GeoTiff loads and saves data to a local geoiff file. The
+underlying functionality is supported by rioxarray and xarray. A read geotiff file
+returns a xarray.DataArray object.
+"""
+
+from copy import deepcopy
+from pathlib import PurePosixPath
+from typing import Any, Dict
+
+import fsspec
+import rioxarray as rxr
+import xarray
+from kedro.io.core import Version, get_filepath_str, get_protocol_and_path
+
+from kedro_datasets._io import AbstractVersionedDataset, DatasetError
+
+
+class GeoTiffDataset(AbstractVersionedDataset[xarray.DataArray, xarray.DataArray]):
+    """``GeoTiffDataset``  loads and saves geotiff files and reads them as xarray
+    DataArrays.
+
+
+    .. code-block:: yaml
+
+        avalanches:
+          type: xarray.GeoTiffDataset
+          filepath: avalanches.tif
```

Comment:
Could you also add an example how to use this with the python api? Like we have here: https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/kedro_datasets/pandas/csv_dataset.py#L52-L67

Reply:
done